File: android/app/src/main/java/ai/oriveo/community/ui/component/LocalModelRuntimeLabel.kt
package ai.oriveo.community.ui.component

import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.res.stringResource
import ai.oriveo.community.R
import ai.oriveo.community.core.model.AIModel
import ai.oriveo.community.core.model.LocalModelLoadState
import ai.oriveo.community.core.model.ModelExecutionLocality
import ai.oriveo.community.ui.theme.OriveoTheme

@Composable
fun LocalModelRuntimeLabel(model: AIModel) {
    val resource = when {
        model.executionLocality == ModelExecutionLocality.ProxiedCloud -> R.string.local_model_via_ollama_cloud
        model.localLoadState == LocalModelLoadState.Loading -> R.string.local_model_loading
        model.localLoadState == LocalModelLoadState.Unloaded -> R.string.local_model_unloaded
        model.localLoadState == LocalModelLoadState.Unknown -> R.string.local_model_state_unknown
        else -> null
    }
    resource?.let {
        Text(
            text = stringResource(it),
            style = OriveoTheme.typography.caption,
            color = if (model.executionLocality == ModelExecutionLocality.ProxiedCloud) OriveoTheme.colors.warning else OriveoTheme.colors.textSecondary,
        )
    }
}

File: ios/Oriveo/Oriveo/Features/Providers/LocalModelRuntimeLabel.swift
import SwiftUI

struct LocalModelRuntimeLabel: View {
    let model: AIModel

    var body: some View {
        if let label {
            Text(label)
                .font(OriveoTheme.Typography.footnote)
                .foregroundStyle(model.executionLocality == .proxiedCloud ? OriveoTheme.Palette.warning : OriveoTheme.Palette.textSecondary)
                .accessibilityLabel(label)
        }
    }

    private var label: String? {
        if model.executionLocality == .proxiedCloud {
            return L10n.tr("Via Ollama Cloud", table: .providers)
        }
        switch model.localLoadState {
        case .loading: return L10n.tr("Loading", table: .providers)
        case .unloaded: return L10n.tr("Unloaded", table: .providers)
        case .unknown: return L10n.tr("State unknown", table: .providers)
        case .loaded, .none: return nil
        }
    }
}
